post-push-kep-manifest job fails: go.mod uses go 1.26.0 but gcloud-in-go image has outdated Go

The `post-push-kep-manifest` Prow postsubmit job is failing on every merge to `kubernetes/enhancements` master.

**Build:** https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/post-push-kep-manifest/2077066934481850368

**Error:**
```
go: errors parsing go.mod:
go.mod:3: invalid go version '1.26.0': must match format 1.23
make: *** [Makefile:88: tools] Error 1
```

**Root cause:**
The job uses image `gcr.io/k8s-staging-test-infra/gcloud-in-go:v20230111-cd1b3caf9c` (built Jan 2023) which bundles an old Go binary that cannot parse the `go 1.26.0` directive in [`kubernetes/enhancements/go.mod`](https://github.com/kubernetes/enhancements/blob/master/go.mod#L3). Only two stale tags of `gcloud-in-go` exist — both from 2022/2023.

The presubmit jobs ([config](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/enhancements/enhancements-presubmit.yaml)) work fine because they use `golang:1.26` directly, but the postsubmit needs `gcloud-in-go` for `gsutil` (GCS upload).

**Fix options:**

1. **Rebuild `gcloud-in-go` with Go 1.26:** Bump `FROM golang:1.22` to `FROM golang:1.26` in [`images/gcloud-in-go/Dockerfile`](https://github.com/kubernetes/test-infra/blob/master/images/gcloud-in-go/Dockerfile), then update the job config at [`config/jobs/kubernetes/sig-k8s-infra/trusted/enhancements-postsubmit.yaml`](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-k8s-infra/trusted/enhancements-postsubmit.yaml) to use the new image tag.

2. **Switch postsubmit to `golang:1.26` + install gcloud SDK at runtime:** Replace the image with `public.ecr.aws/docker/library/golang:1.26` and add `google-cloud-sdk` installation to the run command.

**Impact:** `keps.json` at `gs://k8s-keps/keps.json` has not been updated since June 25, 2026 — meaning the `path` field from [kubernetes/enhancements#6225](https://github.com/kubernetes/enhancements/pull/6225) is missing from all KEP entries. This causes ~30 WARN-level 404s on the contributor-site KEP pages ([kubernetes/contributor-site#804](https://github.com/kubernetes/contributor-site/pull/804)).

--- a/config/jobs/kubernetes/sig-k8s-infra/trusted/enhancements-postsubmit.yaml
+++ b/config/jobs/kubernetes/sig-k8s-infra/trusted/enhancements-postsubmit.yaml
@@ -8,7 +8,7 @@ postsubmits:
       max_concurrency: 1
       spec:
         containers:
-        - image: gcr.io/k8s-staging-test-infra/gcloud-in-go:v20230111-cd1b3caf9c
+        - image: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20260205-38cfa9523f
           command:
             - sh
             - "-c"

